src/record_layout.zig

```diff
@@ -0,0 +1,459 @@
+        // 0084, 0085, 0086.
+        self.size_bits = std.math.max(self.size_bits, offset_bits + size_bits);
+        self.aligned_bits = std.math.max(self.aligned_bits, fld_align_bits);
+
+        fld.layout.offset_bits = offset_bits;
+        fld.layout.size_bits = size_bits;
+    }
+
+    fn layoutBitField(
+        self: *SysVContext,
+        comp: *const Compilation,
+        fld: *Field,
+        fld_attrs: ?[]const Attribute,
+        fld_layout: TypeLayout,
+    ) void {
+        const bit_width = fld.bit_width.?;
+
+        const ty_size_bits = fld_layout.size_bits;
+        var ty_fld_algn_bits: u32 = fld_layout.field_alignment_bits;
+
+        // Some targets ignore the alignment of the underlying type when laying out
+        // non-zero-sized bit-fields. See test case 0072. On such targets, bit-fields never
+        // cross a storage boundary. See test case 0081.
+        if (comp.ignoreNonZeroSizedBitfieldTypeAlignment()) {
+            ty_fld_algn_bits = 1;
+        }
+
+        if (bit_width <= 0) {
+            // Some targets have a minimum alignment of zero-sized bit-fields. See test case
+            // 0074.
+            if (comp.minZeroWidthBitfieldAlignment()) |target_align| {
+                ty_fld_algn_bits = std.math.max(ty_fld_algn_bits, target_align);
+            }
+        }
+
+        // __attribute__((packed)) on the record is identical to __attribute__((packed)) on each
+        // field. See test case 0067.
+        const attr_packed = self.attr_packed or isPacked(fld_attrs);
+        const has_packing_annotation = attr_packed or self.max_field_align_bits != null;
+
+        const annotation_alignment: u32 = annotationAlignment(fld_attrs) orelse 1;
+
+        var first_unused_bit: u64 = if (self.is_union) 0 else self.size_bits;
+        var field_align_bits: u64 = 1;
+
+        if (bit_width == 0) {
+            field_align_bits = std.math.max(ty_fld_algn_bits, annotation_alignment);
+        } else if (comp.langopts.emulate == .gcc) {
+            // On GCC, the field alignment is at least the alignment requested by annotations
+            // except as restricted by #pragma pack. See test case 0083.
+            field_align_bits = annotation_alignment;
+            if (self.max_field_align_bits) |max_bits| {
+                field_align_bits = std.math.min(annotation_alignment, max_bits);
+            }
+
+            // On GCC, if there are no packing annotations and
+            // - the field would otherwise start at an offset such that it would cross a
+            //   storage boundary or
+            // - the alignment of the type is larger than its size,
+            // then it is aligned to the type's field alignment. See test case 0083.
+            if (!has_packing_annotation) {
+                var start_bit = std.mem.alignForwardGeneric(u64, first_unused_bit, field_align_bits);
+
+                var does_field_cross_boundary = start_bit % ty_fld_algn_bits + bit_width > ty_size_bits;
```

Comment:
```suggestion
                const does_field_cross_boundary = start_bit % ty_fld_algn_bits + bit_width > ty_size_bits;
```
`var` instead of `const`, this appears in multiple places so please review usage of `var` everywhere.